src/tls.c

```diff
@@ -819,6 +820,43 @@ static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
     return ret;
 }
 
+static int connTLSWritev(connection *conn_, const struct iovec *iov, int iovcnt) {
+    if (iovcnt == 1) return connTLSWrite(conn_, iov[0].iov_base, iov[0].iov_len);
+
+    /* Accumulate the amount of bytes of each buffer and check if it exceeds NET_MAX_WRITES_PER_EVENT. */
+    size_t iov_bytes_len = 0;
+    for (int i = 0; i < iovcnt; i++) {
+        iov_bytes_len += iov[i].iov_len;
+        if (iov_bytes_len > NET_MAX_WRITES_PER_EVENT) break;
+    }
+
+    /* The amount of all buffers is greater than NET_MAX_WRITES_PER_EVENT, 
+     * which is not worth doing so much memory copying to reduce system calls,
+     * therefore, invoke connTLSWrite() multiple times to avoid memory copies. */
+    if (iov_bytes_len > NET_MAX_WRITES_PER_EVENT) {
+        size_t iov_bytes_len = 0;
+        for (int i = 0; i < iovcnt; i++) {
+            size_t sent = connTLSWrite(conn_, iov[i].iov_base, iov[i].iov_len);
+            if (sent <= 0) return iov_bytes_len > 0 ? iov_bytes_len : sent;
+            iov_bytes_len += sent;
+            if (sent != iov[i].iov_len) break;
+        }
+        return iov_bytes_len;
+    }
```

Comment:
Sorry if i wasn't clear enough. 

`iov_bytes_len` represent the total amount of data referenced in iov[]. In this block section we would like to use a distinct variable than `iov_bytes_len` which represents the the total amount of data that actually sent. Please use another name, maybe `iov_bytes_sent` or alike.

Reply:
Done.